tests/test_mrf_size.py

```diff
@@ -0,0 +1,120 @@
+        # ASSERT
+        self.assertTrue(os.path.exists(vrt_path))
+        tree = ET.parse(vrt_path)
+        root = tree.getroot()
+
+        # 1. Check VRT dimensions (should be 2x1 pixels, one for each tile)
+        self.assertEqual(root.tag, "VRTDataset")
+        self.assertEqual(root.get("rasterXSize"), "2")
+        self.assertEqual(root.get("rasterYSize"), "1")
+
+        # 2. Check GeoTransform (should be scaled by pagesize)
+        # Original pixel res: x=1, y=-1. Scaled by 512 -> x=512, y=-512
+        geotransform = root.find("GeoTransform").text
+        self.assertEqual(geotransform, "0.0,512.0,0,1024.0,0,-512.0")
+
+        # 3. Check VRTRasterBand properties
+        band = root.find("VRTRasterBand")
+        self.assertIsNotNone(band)
+        self.assertEqual(band.get("dataType"), "UInt32")
+        self.assertEqual(band.find("SourceFilename").text, "test.idx")
+        
+        # ImageOffset should be 12 to read the lower 4 bytes of the 8-byte size field
+        self.assertEqual(band.find("ImageOffset").text, "12") 
+        # PixelOffset is 16 bytes (size of one index record)
+        self.assertEqual(band.find("PixelOffset").text, "16")
+        # LineOffset is 16 * rasterXSize * num_bands = 16 * 2 * 1 = 32
+        self.assertEqual(band.find("LineOffset").text, "32")
+        self.assertEqual(band.find("ByteOrder").text, "MSB")
+
+    def test_vrt_creation_multi_band(self):
+        """Test VRT generation for a 3-band, single-tile MRF."""
+        # ARRANGE
+        mrf_path = os.path.join(self.test_dir, "multiband.mrf")
+        vrt_path = os.path.join(self.test_dir, "multiband_size.vrt")
+        self._create_full_mrf_xml(mrf_path, xsize=512, ysize=512, channels=3, pagesize=512)
+
+        # ACT
+        sys.argv = ["mrf_size.py", mrf_path]
+        mrf_size.main()
+        
+        # ASSERT
+        self.assertTrue(os.path.exists(vrt_path))
+        tree = ET.parse(vrt_path)
+        root = tree.getroot()
+        
+        # 1. VRT should be 1x1 pixels
+        self.assertEqual(root.get("rasterXSize"), "1")
+        self.assertEqual(root.get("rasterYSize"), "1")
+
+        # 2. There should be 3 raster bands
+        bands = root.findall("VRTRasterBand")
+        self.assertEqual(len(bands), 3)
+
+        # 3. Check offsets for each band
+        # PixelOffset and LineOffset are 16 * num_bands = 48
+        # ImageOffset increments by 16 for each band
+        self.assertEqual(bands[0].find("ImageOffset").text, "12") # 12 + 16 * 0
+        self.assertEqual(bands[0].find("PixelOffset").text, "48")
+        self.assertEqual(bands[0].find("LineOffset").text, "48")
+
+        self.assertEqual(bands[1].find("ImageOffset").text, "28") # 12 + 16 * 1
+        self.assertEqual(bands[2].find("ImageOffset").text, "44") # 12 + 16 * 2
+
+    def test_vrt_default_pagesize(self):
+        """Test that a default PageSize of 512 is used when the tag is absent."""
+        # ARRANGE: Create a 1024x1024 MRF without a <PageSize> tag
```

Comment:
Not sure why 1024 is used here when we're testing the default of 512.

Reply:
Is that really how it works? My understanding is that rasterXSize and rasterYSize is the output size of the virtual raster in pixels, not the tile dimensions.

Reply:
That's what mrf_size does, it generates a raster where each pixel corresponds to a input tile, with the pixel value being the size of the matching tile. A 1024x1024 MRF with a tile of 512x512 will generate a 2x2 pixel raster output.

Reply:
I haven't used this in a long time, it's a neat tool to visually identify issues in a very large MRF. The output is graphic, georeferenced and can be vied in any GIS.

Reply:
Ah, interesting. That makes more sense now. I haven't tried the tool before but do see where it can be useful.